### DISCUSSION TITLE
Labgrid typing package for mypy?

### DISCUSSION THREAD

**Discussion by DeltaFoxtrot01** [labgrid-project/labgrid#1473] [General]
Hello,

Currently, I am implementing stricter typing in a Python project I am working on. I noticed that a portion of the package does not work with strict typing. I tried looking for a typing package for Labgrid, and I could not find one. I noticed that there is a PR related to this, but the amount of work seems to have just been some small contribution to it (link for the PR -> https://github.com/labgrid-project/labgrid/pull/823). Am I missing something? Is there any plan to make this available?

**Comment by jluebbe** [comment]
The gRPC migration (#1469) uses some more type annotations. We'd also like to use mypy to check labgrid itself, but the last time I looked at that (~2021), mypy didn't understand our Resource/Driver binding mechanism. That could probably be fixed by someone experienced with mypy, though.

So, while we don't have any concrete plans for more mypy annotations, we'd definitely appreciate contributions towards that goal.
